app/scripts/metamask-controller.js

```diff
@@ -2435,6 +2456,11 @@ export default class MetamaskController extends EventEmitter {
       } else {
         this.txController.stopIncomingTransactionPolling();
       }
+
+      this.controllerMessenger.publish(
```

Comment:
What are your thoughts on doing this inside of the PreferencesController constructor? That way when we use PreferencesController from `core`, we don't have to remember to remove this line.

Reply:
That's a sensible idea. Applied here: https://github.com/MetaMask/metamask-extension/commit/1d47a885d21bd6a806437e4ed88380940abfaf90 (the diff is weird because I squashed this into an existing commit instead of doing a separate revert)

Reply:
I'm getting the error below in multiple tests. Not sure what the issue is but restoring the subscription to happen here fixed it. Adding a TODO for the preferences-controller migration for now.

```output
<testcase name="Import flow allows importing multiple tokens from search" time="9.317" classname="allows importing multiple tokens from search">
      <failure message="Expected values to be strictly equal:

1 !== 4

  (Ran on CircleCI Node 2 of 20, Job test-e2e-chrome)" type="AssertionError">
```

Reply:
I think I see what the issue might be. The publish call is `this.controllerMessenger.publish` here but `this.messagingSystem?.publish` inside the constructor, so it won't execute if the `messenger` is undefined. 

I tried creating and passing in messengers to all of the preferences-controller constructors in the tests, but the e2e tests fail. It seems like there's something else that needs fixing that I can't find. I think we could leave this as a TODO for now?